.github/workflows/library_interop_tests.yml

```diff
@@ -0,0 +1,186 @@
+# This workflow performs interoperability tests across the supported runtimes of the MPL.
+name: Library Interoperability Tests
+
+on:
+  workflow_call:
+    inputs:
+      dafny:
+        description: "The Dafny version to run"
+        required: true
+        type: string
+
+jobs:
+  generateEncryptVectors:
+    strategy:
+      matrix:
+        library: [TestVectorsAwsCryptographicMaterialProviders]
+        os: [
+            # TODO just test on mac and ubuntu for now
+            # windows-latest,
+            ubuntu-latest,
+            macos-latest,
+          ]
+        language: [java, net]
+        dotnet-version: ["6.0.x"]
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Support longpaths on Git checkout
+        run: |
+          git config --global core.longpaths true
+
+      - uses: actions/checkout@v3
+      #  Not all submodules are needed.
+      # We manually pull the submodule we DO need.
+      - run: git submodule update --init libraries
+
+      # Set up runtimes
+      - name: Setup .NET Core SDK ${{ matrix.dotnet-version }}
+        if: matrix.language == 'net'
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+
+      - name: Setup Java 17
+        if: matrix.language == 'java'
+        uses: actions/setup-java@v3
+        with:
+          distribution: "corretto"
+          java-version: 17
```

Comment:
CI TestVectors run over every supported java version, for interop I decided 17 was enough, I didn't want to add additional complexity and overkill to these tests

Reply:
Yeah, I am all for only testing one Java.
I just think the one Java should be the oldest Java.
Since Java is always (?) forwards compatible,
but often not backwards compatible.